src/cgutils.cpp

```diff
@@ -1958,20 +1955,18 @@ static Value *box_union(jl_codectx_t &ctx, const jl_cgval_t &vinfo, const SmallB
     if (skip.size() > 0 && skip[0]) {
         // skip[0] specifies where to return NULL or the original pointer
         // if the value was not handled above
```

Comment:
need to update this comment, since you're using a different sentinel value now

Reply:
I'm not sure what that comment was supposed to mean in the first place. Was there any specific reason you wanted to return null here?

Reply:
It's a sentinel value. You also changed the code that reads this, so you just need to update the comment reflect your new design.

The case I think you need to be care of is that, in the old code, I think it could reinterpret a boxed value before converting it to a union, resulting in the type of `val.gcroot` diverging from `val.tindex`.

Reply:
I don't think my code here was correct - at least I didn't understand what it was doing properly. I've refactored this a bit and changed the sentinel value back to NULL. Also added some comments that would have helped me.